src/System.Diagnostics.TraceSource/tests/TraceClassTests.cs

```diff
@@ -1,122 +1,373 @@
+            Assert.Equal(1, listener.GetCallCount(Method.Fail));
+        }
+
+        [Fact]
+        public void Assert2Test()
+        {
+            var listener = new TestTraceListener();
+            var text = new TestTextTraceListener();
+            Trace.Listeners.Add(listener);
+            Trace.Listeners.Add(text);
+            Trace.Assert(true, "Message");
+            Assert.Equal(0, listener.GetCallCount(Method.WriteLine));
+            Assert.Equal(0, listener.GetCallCount(Method.Fail));
+            text.Flush();
+            Assert.DoesNotContain("Message", text.Output);
+            Trace.Assert(false, "Message");
+            Assert.Equal(0, listener.GetCallCount(Method.WriteLine));
+            Assert.Equal(1, listener.GetCallCount(Method.Fail));
+            text.Flush();
+            Assert.Contains("Message", text.Output);
+        }
+
+        [Fact]
+        public void Assert3Test()
+        {
+            var listener = new TestTraceListener();
+            var text = new TestTextTraceListener();
+            Trace.Listeners.Add(listener);
+            Trace.Listeners.Add(text);
+            Trace.Assert(true, "Message", "Detail");
+            Assert.Equal(0, listener.GetCallCount(Method.WriteLine));
+            Assert.Equal(0, listener.GetCallCount(Method.Fail));
+            text.Flush();
+            Assert.DoesNotContain("Message", text.Output);
+            Trace.Assert(false, "Message", "Detail");
+            Assert.Equal(0, listener.GetCallCount(Method.WriteLine));
+            Assert.Equal(1, listener.GetCallCount(Method.Fail));
+            text.Flush();
+            Assert.Contains("Message", text.Output);
+            Assert.Contains("Detail", text.Output);
+        }
+
+        [Fact]
+        public void WriteTest()
+        {
+            var listener = new TestTextTraceListener();
+            Trace.Listeners.Add(listener);
+            Trace.Write("Message", "Category");
+            Trace.Flush();
+            Assert.Equal("Category: Message", listener.Output);
+        }
+
+        [Fact]
+        public void WriteObject1Test()
+        {
+            var listener = new TestTextTraceListener();
+            Trace.Listeners.Add(listener);
+            Trace.Write((Object)"Text");
+            listener.Flush();
+            Assert.Equal("Text", listener.Output);
+        }
+
+        [Fact]
+        public void WriteObject2Test()
+        {
+            var listener = new TestTextTraceListener();
+            Trace.Listeners.Add(listener);
+            Trace.Write((Object)"Message", "Category");
+            Trace.Flush();
+            Assert.Equal("Category: Message", listener.Output);
+        }
+
+        [Fact]
+        public void WriteLineObjectTest()
+        {
+            var listener = new TestTextTraceListener();
+            Trace.Listeners.Add(listener);
+            Trace.WriteLine((Object)"Text");
+            listener.Flush();
+            Assert.Equal("Text\r\n", listener.Output);
```

Comment:
I suspect you are right. I tried to write this as Environment.NewLine but the compiler was complaining about resolving Environment for some reason. I meant to come back and investigate further, but I guess I forgot. The two existing unit tests had hard-coded 
"/r/n" as well, not that that is an excuse.